src/System/Hapistrano/Commands/Internal.hs

```diff
@@ -190,16 +190,51 @@ instance Command (Find File) where
   parseResult Proxy = mapMaybe (parseAbsFile . trim) . lines
 
 -- | @touch@.
-data Touch =
+newtype Touch =
   Touch (Path Abs File)
 
 instance Command Touch where
   type Result Touch = ()
   renderCommand (Touch path) = formatCmd "touch" [Just (fromAbsFile path)]
   parseResult Proxy _ = ()
 
+newtype CheckExists =
+  CheckExists
+  (Path Abs File) -- ^ The absolute path to the file you want to check for existence
+
+instance Command CheckExists where
+  type Result CheckExists = Bool
+  renderCommand (CheckExists path) = 
+    "([ -r " <> fromAbsFile path <> " ] && echo True) || echo False"
```

Comment:
Hey @DavidMazarro Is `-r` compatible with the shells that we support (`zsh` and `bash`)?

Reply:
I'll admit I didn't initially check, so thanks for the heads-up. Yes, it works on both `zsh` and `bash`.